src/decompression.jl

```diff
@@ -354,12 +382,28 @@ function decompress_single_color!(
 end
 
 function decompress!(A::SparseMatrixCSC, B::AbstractMatrix, result::ColumnColoringResult)
-    (; compressed_indices) = result
+    (; compressed_indices, allow_denser) = result
     S = result.bg.S2
-    check_same_pattern(A, S)
+    check_same_pattern(A, S; allow_denser)
     nzA = nonzeros(A)
-    for k in eachindex(nzA, compressed_indices)
-        nzA[k] = B[compressed_indices[k]]
+    if nnz(A) == nnz(S)
+        for k in eachindex(compressed_indices)
+            nzA[k] = B[compressed_indices[k]]
+        end
+    else  # nnz(A) > nnz(S)
+        rvA, rvS = rowvals(A), rowvals(S)
+        shift = 0
+        for j in axes(S, 2)
+            for k in nzrange(S, j)
+                i = rvS[k]
+                while (k + shift) < A.colptr[j] || rvA[k + shift] < i
+                    nzA[k + shift] = zero(eltype(A))
```

Comment:
Do we want to fill the other nonzeros with 0?
Maybe the other coefficients are updated before the call to `decompress!` and we overwrite them.
At least we should document what we do on the nonzeros entries not that are not in the sparsity pattern of `S`.
I imagine a case where we make Jacobian full column rank with a block `αI` below `J` that is never updated.

Example: https://jso.dev/QRMumps.jl/dev/api/#QRMumps.qrm_shifted_spmat

Reply:
I suggest to document that `decompress!` and `decompress_single_color!` don't modify the nonzeros except the ones expected during the decompression.

Reply:
Yeah we fill them with zeros, it is the most logical choice. `decompress!` means "make the target equal to the Jacobian"